guests/eth-block/src/main.rs

```diff
@@ -26,10 +26,15 @@ pub fn main() {
     // Read the input previous block and transaction data
     let input = env::read();
     // Build the resulting block
-    let (header, state) = EthereumStrategy::build_from(&ETH_MAINNET_CHAIN_SPEC, input)
+    let mut output = EthereumStrategy::build_from(&ETH_MAINNET_CHAIN_SPEC, input)
         .expect("Failed to build the resulting block");
-    // Output the resulting block's hash to the journal
-    env::commit(&header.hash());
+    // Abridge successful construction results
+    if let Some(trie_root) = output.compress_state() {
```

Comment:
`compress_state` replaces the state in the output with just its root and returns the original state.

In our current state/preflight code we use the following naming scheme:
- ..._root denotes the root hash
-- ..._root_node denotes the root MptNode with all its children

Thus, this should here be called either `state_root_node` or just `state` as it is not just the root hash